skrub/datasets/tests/test_utils.py

```diff
@@ -1,18 +1,54 @@
+import shutil
+import tempfile
 from pathlib import Path
-from unittest import mock
 
+from skrub.datasets._utils import get_data_dir, get_data_home
 
-@mock.patch("os.path.dirname")
-def test_get_data_dir(mock_os_path_dirname):
+
+def test_get_data_dir():
     """
     Tests function ``get_data_dir()``.
     """
-    from skrub.datasets._utils import get_data_dir
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        tmpdirpath = Path(tmpdirname)
+        assert get_data_dir(data_home=tmpdirpath) == tmpdirpath
+
+        assert get_data_dir(name="tests", data_home=tmpdirpath) == tmpdirpath / "tests"
+
+
+def test_get_data_home_str():
+    """
+    Test function for ``get_data_home()`` when `data_home` is a string.
+    """
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        tmpdirpath = Path(tmpdirname)
 
-    expected_return_value_default = Path("/user/directory/data").absolute()
+        # get_data_home will point to a pre-existing folder
+        assert tmpdirpath.exists()
+        data_home = get_data_home(data_home=tmpdirpath)
+        assert data_home == tmpdirpath
+        assert data_home.exists()
+
+        # if the folder is missing it will be created
+        shutil.rmtree(data_home)
+        assert not data_home.exists()
+        assert not tmpdirpath.exists()
+        data_home = get_data_home(data_home=tmpdirpath)
+        assert data_home.exists()
+
+
+def test_get_data_home_None():
+    """
+    Test function for ``get_data_home()`` with `data_home` set to `None`.
+    """
+    # get path of the folder 'skrub_data' in the user home folder
+    user_path = Path("~") / "skrub_data"
 
-    mock_os_path_dirname.return_value = expected_return_value_default.parent
-    assert get_data_dir() == expected_return_value_default
+    # if the folder does not exist, create it, and clear it
```

Comment:
I would make the following test with to pass if the folder was not existing.

```python
# We should take care of not deleting the folder if our user
# already cached some data
is_already_existing = user_path.exists()

data_home = get_data_home(data_home=None)
assert data_home == user_path
assert data_home.exists()

if no is_already_exist:
    # In case the folder was not existing, the previous
    # call should have created the folder. Check that we get
    # the proper path to the folder if it already exists.
    data_home = get_data_home(data_home=None)
    assert data_home == user_path
    assert data_home.exists()

if not is_already_existing:
    shutil.rmtree(user_path)
```